src/components/display/fewlines/IdentityValidationForm.tsx

```diff
@@ -0,0 +1,78 @@
+import { useRouter } from "next/router";
+import React from "react";
+import styled from "styled-components";
+
+import { IdentityTypes } from "../../../@types/Identity";
+import AlertBar from "./AlertBar";
+
+const IdentityValidationForm: React.FC<{ type: IdentityTypes }> = ({
+  type,
+}) => {
+  const [validationCode, setValidationCode] = React.useState("");
+  const router = useRouter();
+
+  return (
+    <>
+      <Form method="post">
+        <Input
+          type="text"
+          name="value"
+          placeholder="012345"
+          value={validationCode}
+          onChange={(event) => setValidationCode(event.target.value)}
+        />
+        <SendInput type="submit" value={`Confirm ${type}`} />
+      </Form>
+      <Button onClick={() => router.push("/account/logins/")}>
+        Discard all changes
+      </Button>
+      <p>Didn&apos;t receive code?</p>
+      <Button>Resend confirmation code</Button>
+      <AlertBar type={type} />
+    </>
+  );
+};
+
+export default IdentityValidationForm;
+
+export const Button = styled.button`
+  padding: 0.5rem;
+  margin-right: 1rem;
+  border-radius: ${({ theme }) => theme.radii[0]};
+  background-color: transparent;
+  color: ${({ theme }) => theme.colors.green};
+  transition: ${({ theme }) => theme.transitions.quick};
+  font-weight: ${({ theme }) => theme.fontWeights.medium};
+  &:hover {
+    cursor: pointer;
+    background-color: ${({ theme }) => theme.colors.green};
+    color: ${({ theme }) => theme.colors.contrastCopy};
+  }
+  &:active,
+  &:focus {
+    outline: none;
+    background-color: ${({ theme }) => theme.colors.green};
+    color: ${({ theme }) => theme.colors.contrastCopy};
+  }
+`;
+
+const Form = styled.form`
+  display: flex;
+  align-items: center;
+`;
+
+const Input = styled.input`
+  border: ${({ theme }) => `${theme.colors.blacks[0]} ${theme.borders.thin}`};
+  border-radius: ${({ theme }) => theme.radii[0]};
+  padding: 0.5rem;
+
+  &:active,
+  &:focus {
+    outline: none;
+  }
+`;
+
+const SendInput = styled.input`
+  color: ${({ theme }) => theme.colors.green};
+  padding: 0.25em 1em;
```

Comment:
I would have say use the theme spacing, but we need to change the theme to better handle small spacing.

Reply:
OK I ignore this for now